lib/upload.js

```diff
@@ -531,6 +534,40 @@ class BaseUpload {
       return
     }
 
+    if (this.options.enableChecksum) {
+      const optionsReq = this._openRequest('OPTIONS', this.options.endpoint)
+      this._sendRequest(optionsReq)
+        .then((res) => {
+          const isChecksumSupported = res
+            .getHeader('Tus-Extension')
+            .split(',')
+            .includes('checksum')
+          if (!isChecksumSupported) {
+            log('Checksum is not supported by this tus server')
+            return
+          }
+          const supportedAlgorithms = res.getHeader('Tus-Checksum-Algorithm')
+          if (!supportedAlgorithms) {
+            throw new Error(
+              'tus: checksum is enabled, but Tus-Checksum-Algorithm is not defined by tus server',
+            )
+          }
+          let Checksum
+          if (window) {
```

Comment:
We handle code difference between Node.js and Browsers APIs using dependency injections in two different `index.js` files. See https://github.com/tus/tus-js-client/blob/1db985ac2a1531184a26e3e989de4a2d2f90a823/lib/node/index.js#L6-L18 and https://github.com/tus/tus-js-client/blob/1db985ac2a1531184a26e3e989de4a2d2f90a823/lib/browser/index.js#L6-L18. Let's do this here as well to avoid these requires in here.